crates/config/src/sections/upstream_oauth2.rs

```diff
@@ -497,6 +518,17 @@ pub struct Provider {
     #[serde(default)]
     pub fetch_userinfo: bool,
 
+    /// Expected signature for the JWT payload returned by the userinfo
+    /// endpoint.
+    ///
+    /// If null, the response is expected to be an unsigned JSON payload.
+    /// Defaults to `RS256`.
+    #[serde(
+        default = "signed_response_alg_default",
+        skip_serializing_if = "is_signed_response_alg_default"
+    )]
+    pub userinfo_signed_response_alg: Option<JsonWebSignatureAlg>,
```

Comment:
In contrast, by default most userinfo endpoints use the plain JSON response, so here the default should be `None`
```suggestion
    /// Expected signature for the JWT payload returned by the userinfo
    /// endpoint.
    ///
    /// If null, the response is expected to be an unsigned JSON payload.
    /// Defaults to `null`.
    #[serde(
        default,
        skip_serializing_if = "Option::is_none"
    )]
    pub userinfo_signed_response_alg: Option<JsonWebSignatureAlg>,
```